From 05cf7c9a44e1c1151ce6777d7702d0e70fa90ff3 Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Sat, 25 Sep 2021 20:17:01 +0300
Subject: [PATCH 065/322] Revert "[FIO toup] arm: spl: flush and disable cache
 before jumping to OPTEE"

This reverts commit f7f19c643c3d0f730db07f1d6ba3b0075847bb06.

Superseded with a commit from the list.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 arch/arm/lib/spl.c | 11 -----------
 common/spl/spl.c   |  3 ++-
 include/spl.h      |  9 ---------
 3 files changed, 2 insertions(+), 21 deletions(-)

diff --git a/arch/arm/lib/spl.c b/arch/arm/lib/spl.c
index 2dc84f07ef3..b2b54f28fc4 100644
--- a/arch/arm/lib/spl.c
+++ b/arch/arm/lib/spl.c
@@ -77,14 +77,3 @@ void __noreturn jump_to_image_linux(struct spl_image_info *spl_image)
 }
 #endif	/* CONFIG_ARM64 */
 #endif
-
-#if CONFIG_IS_ENABLED(OPTEE)
-void __noreturn jump_to_image_optee(struct spl_image_info *spl_image)
-{
-	/* flush and turn off caches before jumping to OPTEE */
-	cleanup_before_linux();
-
-	spl_optee_entry(NULL, NULL, spl_image->fdt_addr,
-			(void *)spl_image->entry_point);
-}
-#endif
diff --git a/common/spl/spl.c b/common/spl/spl.c
index b1196501a7e..e3d84082f44 100644
--- a/common/spl/spl.c
+++ b/common/spl/spl.c
@@ -706,7 +706,8 @@ void board_init_r(gd_t *dummy1, ulong dummy2)
 #if CONFIG_IS_ENABLED(OPTEE)
 	case IH_OS_TEE:
 		debug("Jumping to U-Boot via OP-TEE\n");
-		jump_to_image_optee(&spl_image);
+		spl_optee_entry(NULL, NULL, spl_image.fdt_addr,
+				(void *)spl_image.entry_point);
 		break;
 #endif
 #if CONFIG_IS_ENABLED(OPENSBI)
diff --git a/include/spl.h b/include/spl.h
index cf9e5eaab1e..18a80a31210 100644
--- a/include/spl.h
+++ b/include/spl.h
@@ -404,15 +404,6 @@ int spl_board_boot_device(u32 boot_device);
  */
 void __noreturn jump_to_image_linux(struct spl_image_info *spl_image);
 
-/**
- * jump_to_image_linux() - Jump to OP-TEE OS from SPL
- *
- * This jumps into OP-TEE OS using the information in @spl_image.
- *
- * @spl_image: Image description to set up
- */
-void __noreturn jump_to_image_optee(struct spl_image_info *spl_image);
-
 /**
  * spl_start_uboot() - Check if SPL should start the kernel or U-Boot
  *
-- 
2.34.1

